compiler/rustc_mir/src/const_eval/error.rs

```diff
@@ -193,7 +197,13 @@ impl<'tcx> ConstEvalErr<'tcx> {
                     rustc_session::lint::builtin::CONST_ERR,
                     hir_id,
                     tcx.span,
-                    |lint| finish(lint.build(message), Some(err_msg)),
+                    |lint| {
+                        if is_panic {
```

Comment:
I think something similar needs to be done for the `Report as hard error` case below. Maybe the ` match &self.error` above could produce a `message` and an `Option<String>` for the note, and then you can set the appropriatly once?